File: 0217-contains-duplicate/0217-contains-duplicate.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var containsDuplicate = function(nums) {
  const mp = new Map();
  for (let i = 0; i < nums.length; i++) {
    if (mp.has(nums[i])) {
      return true;
    }
    mp.set(nums[i], 1);
  }
  return false;
};

File: 0217-contains-duplicate/0217-contains-duplicate.go

func containsDuplicate(numbers []int) bool {
	seenNumbers := make(map[int]bool)

	for _, number := range numbers {
		// Check if the number has already been seen
		if seenNumbers[number] {
			return true
		}

		// If the number is not in seenNumbers, add it.
		seenNumbers[number] = true

	}

	// If no duplicates were found, return false
	return false
}